### ISSUE TITLE
airfoil_dp_mpi_cuda doesn't build

### ISSUE THREAD

**Issue by robertsawko** [OP-DSL/OP2-Common#137]
Hi,

I am trying to build OP2 apps to run some tests too, but I can't seem to build all `airfloil` app variant due to some esoteric cmake error, I think. If you can, please advise if there's anything obvious I am doing wrong or that I can fix.

The confusing bit is that despite the warning some variants get built and some don't. For instance `airfoil_tempdats_mpi_cuda` gets build but `airfoil_dp_mpi_cuda` does not.

The cmake line is based pretty much on what you left in the `apps` directory:

```bash
  eval cmake -DCMAKE_INSTALL_PREFIX=$INSTALLDIR \
             -DCMAKE_BUILD_TYPE=Developer \
             -DCMAKE_C_FLAGS_DEVELOPER="-O2 -g -Wall -Werror" \
             -DCMAKE_CXX_FLAGS_DEVELOPER="-O2 -g -Wall -Werror -Wno-long-long" \
             -DCUDA_USE_STATIC_CUDA_RUNTIME=OFF \
             -DAIRFOIL_MESH_FILE=../new_grid.dat \
             $CMAKE_EXTRA_ARGS \
             .. &&
             (make -j$CORES all; make install)
```

I get plenty of errors of this type:
```  Policy CMP0046 is not set: Error on non-existent dependency in
  add_dependencies.  Run "cmake --help-policy CMP0046" for policy details.
  Use the cmake_policy command to set the policy and suppress this warning.

  The dependency target "AIRFOIL_grid" of target "airfoil_tempdats_mpi" does
  not exist.
Call Stack (most recent call first):
  airfoil/airfoil_tempdats/CMakeLists.txt:56 (op2_application)
```

**Comment by robertsawko** [comment]
As always, plot thickens the moment I post something... Seems like the grid warning is a red herring. The real culrpit is here:

```
-- Library INCLUDE_DIRECTORIES not available, skipping application airfoil_sp_mpi_cuda
-- Library INCLUDE_DIRECTORIES not available, skipping application airfoil_sp_mpi_openmp
```

**Comment by robertsawko** [comment]
Ok. I will review this. I can confirm that the slightly older cmake version builds it correctly. Let's leave it open for now,

**Comment by robertsawko** [comment]
I've built it by simply removing

```cmake
# op2_application(airfoil_${VARIANT}_mpi_openmp DEPENDS AIRFOIL_grid LIBS op2_mpi op2_openmp
  #   INCLUDE_DIRECTORIES ${VARIANT} SOURCES ${VARIANT}/airfoil_mpi_op.cpp ${VARIANT}/openmp/airfoil_kernels.cpp
  #   ${VARIANT}/save_soln.h ${VARIANT}/adt_calc.h ${VARIANT}/res_calc.h ${VARIANT}/bres_calc.h ${VARIANT}/update.h)
op2_application(airfoil_${VARIANT}_mpi_openmp DEPENDS AIRFOIL_grid LIBS op2_mpi op2_openmp
     SOURCES ${VARIANT}/airfoil_mpi_op.cpp ${VARIANT}/openmp/airfoil_kernels.cpp
     ${VARIANT}/save_soln.h ${VARIANT}/adt_calc.h ${VARIANT}/res_calc.h ${VARIANT}/bres_calc.h ${VARIANT}/update.h)
```
The `INCLUDE_DIRECTORIES` does not appear to exist in parsing of `op2_application` function.

I will cross check on a few machines and submit a PR.

**Comment by gihanmudalige** [comment]
resolved with #138
